app/views/admin/api_keys.py

```diff
@@ -0,0 +1,225 @@
+# Copyright 2019 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""The admin API key pages.
+
+The way navigation is handled for these pages is a little unintuitive so let me
+explain:
+ - The main entry point to these pages is the list view, which accepts GET
+   requests and returns a view with a) a link to create a new API key (which
+   makes a GET request to the management page) and b) a list of existing pages,
+   each with an "edit" button.
+ - The list view's "edit" buttons make a _POST_ request to send the user to the
+   management page for that key, with the key in a CGI parameter. I'm not
+   exactly sure why that's a POST instead of a GET request, but I would guess
+   it's to keep API keys out of URLs (which would end up in browser histories).
+ - When a key is created or edited through the management page, it makes a POST
+   request to itself, which returns a redirect to itself (which becomes a GET
+   request). Consequently, the POST/GET handlers are the opposite of what you
+   might expect: the POST handler handles the initial display of the form, while
+   the GET handler handles what you see afterwards. The GET handler takes the
+   Datastore key for a log entry (via a CGI param), which it uses to get the API
+   key itself by way of the log entry.
+"""
+
+import django.shortcuts
+import django.utils.translation as t
+from google.appengine.ext import db
+
+import model
+import utils
+import views.admin.base
+
+
+_API_KEY_LENGTH = 16
+
+
+class ApiKeyListView(views.admin.base.AdminBaseView):
+    """The admin API key list view."""
+
+    ACTION_ID = 'admin/api_keys/list'
+
+    def get(self, request, *args, **kwargs):
+        """Serves get requests.
```

Comment:
maybe explain what the method is actually doing? i.e. what is returned (and then you can probably drop the 'Returns' clause).

Reply:
Done.